mlr_backend/run.py

```diff
@@ -1,27 +1,36 @@
-from flask import Flask, jsonify, request
+from flask import Flask, jsonify, request, session
 from flask_cors import CORS
+from io import StringIO
+from flask_caching import Cache
+from kmeans.kmeans import kMeans
+import numpy as np
 
 app = Flask(__name__)
+app.config["CACHE_TYPE"] = "SimpleCache"
+cache = Cache(app)
 CORS(app)
 
 app.secret_key = "2130"
 
 
 @app.route("/upload", methods=["POST"])
 def uploadCSVFile():
-    request_data = request.get_json()
-    print("request data is")
-    print(request_data)
-    # TODO: Upload data
+    input_dataset = request.files['file'].read().decode('utf-8')
+    cache.set("dataset", input_dataset)
     return jsonify({"message": "uploaded successfully"})
 
 
 @app.route("/training", methods=["POST"])
 def trainData():
     request_data = request.get_json()
-    print("request data is")
-    print(request_data)
-    # TODO: Use ML Models to train data and send back useful information
+    models = request_data.get('models')
+    params = request_data.get('params')
+    dataset = cache.get("dataset")
+    dataset_in_np_array = np.genfromtxt(StringIO(dataset),
+                                        delimiter=',', usecols=np.arange(0, 5))
+    if ("KMeans" in models):
+        centroid, clusterAssessment = kMeans(
+            dataset_in_np_array, int(params['Number_of_Clusters']))
```

Comment:
Quick question: Should we include the accuracy calculation right now?

Reply:
Or we can just do a separate PR for that

Reply:
Thanks Tommy for the comment! I am not quite familiar with ML :( Wondering what is the accuracy calculation and how we can do that.

Reply:
As discussed offline, we are moving this calculation to a separate PR as there is quite a bit of work